Makefile

```diff
@@ -38,6 +38,22 @@ superset:
 	# Load some data to play with
 	superset load-examples
 
+update:
```

Comment:
Maybe name this something like `update-npm` and then only do the final step, as the `install` will bump versions in the lockfile:

```
cd superset-frontend
npm install
```

Reply:
Install should only install what's in the lockfile (when given no args). `npm update` will bump versions.

Reply:
I would consider splitting FE and BE related functionality into different commands that can be run independent of one another. Then, we can also have a command that runs both commands, e.g., `update-all: update-be update-fe`.

Reply:
`npm ci` will only install what's in the lockfile. `install` will bump patches

Reply:
`npm ci` will reinstall everything in the lockfile (essentially wiping `node_modules` and then installing fresh packages). `npm install` will instal the missing dependencies in `node_modules`, if there's no lockfile one will be created, if there is a lockfile it will be adhered to.

Reply:
@nytai `npm install` will also bump dependencies if a new version is available that matches the given range. Since we don't pin our dependencies, we could wind up with updates to package-lock.json without any changes to package.json. Depending on if that's what we want, but my opinion would be for package-lock.json updates in PRs only to be there when there's an explicit change in the package.json file. It makes reviews easier. My vote would be to use `npm ci` for updates, even though it takes longer.

Reply:
Ah, I didn't know, though somehow I'm not surprised. I found this issue, apparently it's quite a controversial subject 
https://github.com/npm/npm/issues/17979... I guess it's too much to ask for your package manager to behave predictably 🤦

Reply:
Though, reading through the issue it appears this behavior has been patched in recent versions of `npm@^5.4.2` and it behaves as I have described above
https://github.com/npm/npm/issues/17979#issuecomment-332701215

Reply:
Ha ha.. so I've still been seeing updated package-lock.json files in PRs that shouldn't be there, but.. I tested it out with v 7.6.0 and pulled Superset 1.0.0 and ran `npm install` and it updated the lockfile. :(
Either way, we can leave this as is, but if we do, I think we should be extra vigilant about sneaky tailgaiting package-lock files in PRs. Technically it _should_ do no harm to have them updated, if the package is using semver correctly, but it just creates more ambiguity and room for potential bugs if it's there for no seemingly good reason.

Reply:
I'll still continue to recommend that people use `npm ci` when updating.

Reply:
I've seen the same but I think that's due to slight differences in npm version. I haven't seen actual package version changes in the lockfile, it seems like it's usually formatting changes or updates to the integrity sha. I think it would be valuable to lock down the [version of node & npm we are all using](https://github.com/apache/superset/blob/765e9dd93259fbf2ded9117017704e3fd5cd86d8/superset-frontend/package.json#L58) to a specific version and update these scripts to ensure the correct version of node/npm is being used.

Reply:
Or just use npm ci :)